include/irr/asset/IPipelineLayout.h

```diff
@@ -15,6 +15,20 @@ namespace irr
 namespace asset
 {
 
+//! Push Constant Ranges
+/*
+    Push Constants serve a similar purpose to a Uniform Buffer Object,
+    however they serve as a fast path with regard to data upload from the
+    CPU and data access from the GPU.
+
+    Push Constants are an alternative to an UBO where it performs really poorly,
+    mostly very small and very frequent updates. Examples of which are:
+
+    - Global Object ID
+    - Material/Mesh flags implemented as bits
+    - Unique per DrawCall indices or bit-flags
```

Comment:
state that irrlichtBaW limits push constant size to 128 bytes

Reply:
adjusted in following commit